plio/views.py

```diff
@@ -451,79 +451,94 @@ def metrics(self, request, uuid):
                     "item_type",
                     "question_type",
                     "correct_answer",
+                    "survey",
                 ],
             )
 
             # retain only the responses to items which are questions
-            question_df = df[df["item_type"] == "question"].reset_index(drop=True)
-            num_questions = len(questions)
-
-            def is_answer_correct(row):
-                """Whether the answer corresponding to the given row is correct"""
-                if row["question_type"] in ["mcq", "checkbox"]:
-                    return row["answer"] == row["correct_answer"]
-                return row["answer"] is not None
-
-            # holds the number of questions answered for each viewer
-            num_answered_list = []
-            # holds the number of questions correctly answered for each viewer
-            num_correct_list = []
-
-            user_grouping = question_df.groupby("user_id")
-            for group in user_grouping.groups:
-                # get the responses for a given user
-                group_df = user_grouping.get_group(group)
-
-                # sanity check
-                assert num_questions == len(
-                    group_df
-                ), "Inconsistency in the number of questions"
-
-                num_answered = sum(
-                    group_df["answer"].apply(lambda value: value is not None)
+            dff = df[df["item_type"] == "question"].reset_index(drop=True)
+            question_df = dff[dff.survey.eq(False)].reset_index(drop=True)
+            num_questions = len(question_df)
+
+            if num_questions == 0:
+                return Response(
+                    {
+                        "unique_viewers": num_unique_viewers,
+                        "average_watch_time": average_watch_time,
+                        "isQuesSurvey": True,
```

Comment:
there is an edge case here that will make this incorrect. think about it. Also, for different cases, we have returned all the metrics always at the end at once so that all the keys are present. We should keep that same so that the results always include all the metric keys.